openfl/callbacks/callback_list.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2020-2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+from typing import List
+
+from openfl.callbacks.callback import Callback
+from openfl.callbacks.memory_profiler import MemoryProfiler
+
+
+class CallbackList(Callback):
+    """An ensemble of callbacks.
+
+    This class allows multiple callbacks to be used together, by sequentially
+    calling each callback's respective methods.
+
+    Attributes:
+        callbacks: A list of `openfl.callbacks.Callback` instances.
+        add_memory_profiler: If True, adds a `MemoryProfiler` callback to the list.
+        tensor_db: Optional `TensorDB` instance of the respective participant.
+            If provided, callbacks can access TensorDB for various actions.
+        params: Additional parameters saved for use within the callbacks.
+    """
+
+    def __init__(
+        self,
+        callbacks: List[Callback],
+        add_memory_profiler=False,
+        tensor_db=None,
+        **params,
+    ):
+        super().__init__()
+        self.callbacks = _flatten(callbacks) if callbacks else []
+
+        self._add_default_callbacks(add_memory_profiler)
+
+        self.set_tensor_db(tensor_db)
+        self.set_params(params)
+
+    def set_params(self, params):
+        self.params = params
+        if params:
+            for callback in self.callbacks:
+                callback.set_params(params)
```

Comment:
can there be params which are not applicable to all callbacks but to a specific callback?

Reply:
The intent is to define params during initialization that are visible to all callbacks in the list. For params that are limited to a specific callback, a user may choose to subclass and provide them during `__init__` of the respective class.